.github/workflows/CI.yml

```
@@ -182,3 +182,66 @@ jobs:
         with:
           key: ${{ steps.build-cache.outputs.cache-primary-key }}
           path: ~/.cache
+  release:
+    name: "Publish Pre-Release"
+    runs-on: ubuntu-20.04-16core
+    needs: [test]
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: "1.8"
+          arch: "x64"
+      - uses: julia-actions/cache@v1
+      - name: Check Project version
+        id: version_check
+        shell: julia --color=yes --project {0}
+        run: |
+          using Pkg.Types
+          project = read_project("Project.toml")
+          open(ENV["GITHUB_OUTPUT"], "a") do io
+            println(io, "version=v$(project.version)")
+            println(io, "release_branch=release-v$(project.version)")
+          end
+      - name: Download Artifacts
+        shell: julia --color=yes --project {0}
+        run: |
+          using Pkg
+          Pkg.instantiate()
+          include(joinpath(pwd(), "download_tarballs.jl"))
+          download_tarballs()
+        working-directory: ${{ env.build_dir }}
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+      - name: Update Artifacts.toml
+        shell: julia --color=yes --project {0}
+        run: |
+          using Pkg
+          Pkg.instantiate()
+          include(joinpath(pwd(), "bind_artifacts.jl"))
+          bind_artifacts()
+        working-directory: ${{ env.build_dir }}
+      - name: Commit Changes
+        id: commit
+        run: |
+          git checkout -b ${{ steps.version_check.outputs.release_branch }}
+          git commit -m "Update Artifacts.toml" -- Artifacts.toml
+          echo "commitish=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
```

Shows this output as well as appending to the `$GITHUB_OUTPUT` file
```suggestion
          echo "commitish=$(git rev-parse HEAD)" | tee -a "$GITHUB_OUTPUT"
```